radicle-crypto/src/test/signer.rs

```diff
@@ -62,4 +62,8 @@ impl Signer for MockSigner {
     fn sign(&self, msg: &[u8]) -> Signature {
         self.sk.sign(msg, None).into()
     }
+
+    fn try_sign(&self, msg: &[u8]) -> Result<Signature, SignerError> {
```

Comment:
Would we not mark these ones as `Infallible`?
```suggestion
    fn try_sign(&self, msg: &[u8]) -> Result<Signature, Infallible> {
```

Reply:
Yeah, but that would require an associated error type which then bubbles up across Error types, unless we Box it. Could be worthwhile though?

Reply:
Mmmmm I see. I tend to do the associated error type. Whether it's worth it in this case I guess depends on how often the fallible one is used over the infallible one. Like, if it's mostly fallible then you'll end up changing a lot of the code to handle the boxing.

Reply:
Yeah, do you want to try your hand at it in a subsequent PR?

Reply:
Sure, I've made an issue to remind myself https://github.com/radicle-dev/heartwood/issues/78